google/generativeai/types/retriever_types.py

```diff
@@ -0,0 +1,1237 @@
+_REMOVE = _REMOVE.replace("-", "")  # Don't remove hyphens
+_PATTERN = r"[{}]".format(_REMOVE)  # Create the pattern
+
+Operator = glm.Condition.Operator
+State = glm.Chunk.State
+
+OperatorOptions = Union[str, int, Operator]
+StateOptions = Union[str, int, State]
+
+CreateChunkOptions = Union[Mapping[str, str], tuple[str, str]]
+
+BatchCreateChunkOptions = Union[
+    glm.BatchCreateChunksRequest,
+    list[glm.CreateChunkRequest],
+    Iterable[str],
+    Iterable[CreateChunkOptions],
+]
+
+UpdateChunkOptions = Union[glm.UpdateChunkRequest, Mapping[str, Any], tuple[str, Any]]
+
+BatchUpdateChunksOptions = Union[glm.BatchUpdateChunksRequest, Iterable[UpdateChunkOptions]]
+
+BatchDeleteChunkOptions = Union[list[glm.DeleteChunkRequest], Iterable[str]]
+
+_OPERATOR: dict[OperatorOptions, Operator] = {
+    Operator.OPERATOR_UNSPECIFIED: Operator.OPERATOR_UNSPECIFIED,
+    0: Operator.OPERATOR_UNSPECIFIED,
+    "operator_unspecified": Operator.OPERATOR_UNSPECIFIED,
+    "unspecified": Operator.OPERATOR_UNSPECIFIED,
+    Operator.LESS: Operator.LESS,
+    1: Operator.LESS,
+    "operator_less": Operator.LESS,
+    "less": Operator.LESS,
+    "<": Operator.LESS,
+    Operator.LESS_EQUAL: Operator.LESS_EQUAL,
+    2: Operator.LESS_EQUAL,
+    "operator_less_equal": Operator.LESS_EQUAL,
+    "less_equal": Operator.LESS_EQUAL,
+    "<=": Operator.LESS_EQUAL,
+    Operator.EQUAL: Operator.EQUAL,
+    3: Operator.EQUAL,
+    "operator_equal": Operator.EQUAL,
+    "equal": Operator.EQUAL,
+    "==": Operator.EQUAL,
+    Operator.GREATER_EQUAL: Operator.GREATER_EQUAL,
+    4: Operator.GREATER_EQUAL,
+    "operator_greater_equal": Operator.GREATER_EQUAL,
+    "greater_equal": Operator.GREATER_EQUAL,
+    Operator.NOT_EQUAL: Operator.NOT_EQUAL,
+    5: Operator.NOT_EQUAL,
+    "operator_not_equal": Operator.NOT_EQUAL,
+    "not_equal": Operator.NOT_EQUAL,
+    "!=": Operator.NOT_EQUAL,
+    Operator.INCLUDES: Operator.INCLUDES,
+    6: Operator.INCLUDES,
+    "operator_includes": Operator.INCLUDES,
+    "includes": Operator.INCLUDES,
+    Operator.EXCLUDES: Operator.EXCLUDES,
+    6: Operator.EXCLUDES,
+    "operator_excludes": Operator.EXCLUDES,
+    "excludes": Operator.EXCLUDES,
+    "not in": Operator.EXCLUDES,
+}
+
+_STATE: dict[StateOptions, State] = {
+    State.STATE_UNSPECIFIED: State.STATE_UNSPECIFIED,
+    "0": State.STATE_UNSPECIFIED,
+    "state_unspecifed": State.STATE_UNSPECIFIED,
+    "unspecified": State.STATE_UNSPECIFIED,
+    State.STATE_PENDING_PROCESSING: State.STATE_PENDING_PROCESSING,
+    "1": State.STATE_PENDING_PROCESSING,
+    "pending_processing": State.STATE_PENDING_PROCESSING,
+    "pending": State.STATE_PENDING_PROCESSING,
+    State.STATE_ACTIVE: State.STATE_ACTIVE,
+    "2": State.STATE_ACTIVE,
+    "state_active": State.STATE_ACTIVE,
+    "active": State.STATE_ACTIVE,
+    State.STATE_FAILED: State.STATE_FAILED,
+    "10": State.STATE_FAILED,  # TODO: This is specified as 10 in the proto, should it be 3 or 10?
```

Comment:
10 is correct - this was to leave room for other states in the future and wanting to keep FAILED at the "end".